### PR TITLE
7307 send vector flag true in the editor when number of features is less than 10k

### PR DESCRIPTION
**Pull request by juanignaciosl**
This is the backend side of #7307. It enables vector based on the statistic of row number. Since it's an statistic it's not always up-to-date and might enable vector if the table has recently grown beyond the threshold.

We still need to disable vector if user adds a layer. @xavijam can any frontend take care of it or should I dive into it?

@javitonino CR this, please.

cc @javisantana

### CODE REVIEW

File: app/models/carto/layer.rb [hunk 1]
```diff
@@ -99,6 +99,15 @@ def supports_labels_layer?
       basemap? && options["labels"] && options["labels"]["url"]
     end
 
+    def number_of_features(max = nil)
```
Comment (javitonino) [inline app/models/carto/layer.rb]: Why add a `max` if you are not using it?

**Comment by javitonino** [comment]
I left a minor comment. Implementation seems reasonable. Stats are updated based on number of inserts or other operations (not time). so this should be fairly accurate even if the table grows suddenly. :+1:

File: app/models/carto/layer.rb (hunk 1 above)
Comment (juanignaciosl) [inline app/models/carto/layer.rb]: Ok, I'm stupid. Sorry >_<

**Comment by juanignaciosl** [comment]
Closing for the moment, as this is blocked and it won't be merged anytime soon.

**Pull request closed without merging**